### PR TITLE
Add deepchem

### PR DESCRIPTION
**Pull request by simonbray**
Moved from https://github.com/conda-forge/staged-recipes/pull/10177.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/deepchem/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/deepchem/deepchem/archive/{{ version }}.tar.gz
+  sha256: eec0d4dfa23e234d26d171ef20c62a699052a1bc230bbace341047e0c2673030
+
+build:
+  number: 0
+  script: "PBR_VERSION=${PKG_VERSION} && export PBR_VERSION && {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+  skip: True   # [win or py==27]
```
Comment (scopatz) [inline recipes/deepchem/meta.yaml]: Can this be noarch Python?

Comment (simonbray) [reply recipes/deepchem/meta.yaml]: In principle yes but the windows build fails with `Unsatisfiable dependencies for platform win-64: {'xgboost'}.` 

See: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=119464&view=logs&jobId=240f1fee-52bc-5498-a14a-8361bde76ba0&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=78a0099b-3f13-5ae9-c1f0-24f71d1eac20

Comment (scopatz) [reply recipes/deepchem/meta.yaml]: noarch only needs to build on linux, so you shouldn't worry about the windows failures

Comment (simonbray) [reply recipes/deepchem/meta.yaml]: @scopatz, have updated and the linux build is passing.

**Comment by scopatz** [comment]
Thanks!

**Comment by rbharath** [comment]
@simonbray Thanks for putting this together! This is really exciting to see deepchem added to the condo-forge build :). (Sorry for the slow response; been offline for the last couple weeks). Feel free to tag me if there are any DeepChem issues that pop up in the future

**Comment by jaimergp** [comment]
For those following this conversation, check https://github.com/conda-forge/openmm-feedstock/issues/37 for more information about how to test `openmm` before we move to `main`. The additional ecosystem will be added once that happens (check https://github.com/conda-forge/staged-recipes/pull/13234).

**Pull request merged**